cloudinit/sources/helpers/azure.py

```diff
@@ -33,7 +35,13 @@
 BOOT_EVENT_TYPE = 'boot-telemetry'
 SYSTEMINFO_EVENT_TYPE = 'system-info'
 DIAGNOSTIC_EVENT_TYPE = 'diagnostic'
+COMPRESSED_EVENT_TYPE = 'compressed'
+# Maximum number of bytes of the cloud-init.log file that
+# can be dumped to KVP at once
+MAX_LOG_TO_KVP_LENGTH = 512000
```

Comment:
I don't understand this value,shouldn't  this value be sourcing cloudinit/reporting/handlers.py:HV_KVP_EXCHANGE_MAX_VALUE_SIZE  and multiplying by 1024?

Reply:
This value is based on the collected performance statistics for the cloud-init.file size. It basically indicates how much is the maximum amount of the cloud-init.log file we can afford reporting to KVP with an acceptable performance impact.
I don't think tying this value to the HV_KVP_EXCHANGE_MAX_VALUE_SIZE is the right thing to do here since there is many other factors affecting the performance of how much data we can dump from the cloud-init.log.

Reply:
+1 thanks for that context. If you can add a little bit of info about performance impact of this sizing choice in the comment above it'll help "future us" when we look at this and wonder "why" in the future